packages/playwright-core/src/client/page.ts

```diff
@@ -179,26 +179,22 @@ export class Page extends ChannelOwner<channels.PageChannel> implements api.Page
     this.emit(Events.Page.FrameDetached, frame);
   }
 
-  private _onRoute(route: Route, request: Request) {
+  private async _onRoute(route: Route, request: Request) {
     for (const routeHandler of this._routes) {
       if (!routeHandler.matches(request.url()))
         continue;
       // Immediately deactivate based on |times|.
       if (routeHandler.willExpire())
         this._routes.splice(this._routes.indexOf(routeHandler), 1);
```

Comment:
I think that splicing an array while iterating it will mess up with the iteration. Now that we do not break after first route, this matters.